install.sh

```diff
@@ -748,59 +708,42 @@ _setup_arguments() {
                     CONFIG="${2}" && shift
                 fi
                 ;;
-            --skip-internet-check)
-                SKIP_INTERNET_CHECK=":"
-                ;;
-            -U | --uninstall)
-                UNINSTALL="true"
-                ;;
-            -D | --debug)
-                DEBUG=true
-                export DEBUG
-                ;;
-            *)
-                printf '%s: %s: Unknown option\nTry '"%s -h/--help"' for more information.\n' "${0##*/}" "${1}" "${0##*/}" && exit 1
-                ;;
+            --sh | --posix) INSTALLATION="sh" ;;
+            -q | --quiet) QUIET="_print_quiet" ;;
+            --skip-internet-check) SKIP_INTERNET_CHECK=":" ;;
+            -U | --uninstall) UNINSTALL="true" ;;
+            -D | --debug) DEBUG="true" && export DEBUG ;;
+            *) printf '%s: %s: Unknown option\nTry '"%s -h/--help"' for more information.\n' "${0##*/}" "${1}" "${0##*/}" && exit 1 ;;
         esac
         shift
     done
 
     # 86400 secs = 1 day
     AUTO_UPDATE_INTERVAL="${AUTO_UPDATE_INTERVAL:-432000}"
 
-    if [[ -z ${SHELL_RC} ]]; then
-        printf "No default shell file found, use -s/--shell-rc to use custom rc file\n"
-        exit 1
-    else
-        if ! [[ -f ${SHELL_RC} ]]; then
-            printf "Given shell file ( %s ) does not exist.\n" "${SHELL_RC}"
-            exit 1
-        fi
-    fi
+    [ -z "${SHELL_RC}" ] && printf "No default shell file found, use -s/--shell-rc to use custom rc file\n" && exit 1
 
     _check_debug
 
     return 0
 }
 
 main() {
-    _check_bash_version && _check_dependencies
-    set -o errexit -o noclobber -o pipefail
+    { command -v bash && _check_bash_version && INSTALLATION="bash"; } 2> /dev/null 1>&2
+    _check_dependencies "${?}" && INSTALLATION="${INSTALLATION:-sh}"
 
-    _variables && _setup_arguments "${@}"
+    set -o errexit -o noclobber
 
-    [[ -n ${INTERACTIVE} ]] && _start_interactive
+    _variables && _setup_arguments "${@}"
 
-    if [[ -n ${UNINSTALL} ]]; then
-        if type -a "${COMMAND_NAME}" &> /dev/null; then
-            _uninstall
-        else
+    if [ -n "${UNINSTALL}" ]; then
+        command -v "${COMMAND_NAME}" 2> /dev/null 1>&2 && _uninstall || {
```

Comment:
Codacy found an issue: [Note that A && B || C is not if-then-else. C may run when A is true.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)